sdk/go/pulumi/rpc.go

```diff
@@ -66,7 +66,7 @@ func marshalInputs(props Input) (resource.PropertyMap, map[string][]URN, []URN,
 	depset := map[URN]bool{}
 	pmap, pdeps := resource.PropertyMap{}, map[string][]URN{}
 
-	if props == nil {
+	if props == nil || (reflect.ValueOf(props).Kind() == reflect.Ptr && reflect.ValueOf(props).IsNil()) {
```

Comment:
Let's move the nil pointer check inside this condition and return from there: https://github.com/pulumi/pulumi/pull/4010/files#diff-2372a04ad4c50986c77de67e75de03a9R74-R76